tsl/test/expected/bgw_job_stat_history.out

```diff
@@ -299,6 +299,44 @@ ORDER BY id;
  public      | custom_job_ok | t         | {"only_last_change_is_logged": 1}      |            | 
 (5 rows)
 
+-- Alter other information about the job
+CREATE PROCEDURE custom_job_alter(job_id int, config jsonb) LANGUAGE PLPGSQL AS
+$$
+BEGIN
+  RAISE LOG 'custom_job_alter';
+END
+$$;
+SELECT add_job('custom_job_alter', schedule_interval => interval '1 hour', initial_start := now()) AS job_id_3 \gset
+SELECT test.wait_for_job_to_run(:job_id_3, 1);
+ wait_for_job_to_run 
+---------------------
+ t
+(1 row)
+
+SELECT timezone, fixed_schedule, config, schedule_interval
+FROM alter_job(:job_id_3, timezone => 'America/Sao_Paulo', fixed_schedule => false, config => '{"key": "value"}'::jsonb, schedule_interval => interval '10 min', next_start => now());
+     timezone      | fixed_schedule |      config      | schedule_interval 
+-------------------+----------------+------------------+-------------------
+ America/Sao_Paulo | f              | {"key": "value"} | @ 10 mins
+(1 row)
+
+SELECT test.wait_for_job_to_run(:job_id_3, 2);
+ wait_for_job_to_run 
+---------------------
+ t
+(1 row)
+
+-- Should return two executions, the second will show the changed values
+SELECT job_id, succeeded, data
+FROM _timescaledb_internal.bgw_job_stat_history
+WHERE job_id = :job_id_3
+ORDER BY id;
+ job_id | succeeded |                                                                                                                                                                    data                                                                                                                                                                     
+--------+-----------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+   1002 | t         | {"job": {"owner": "super_user", "proc_name": "custom_job_alter", "scheduled": true, "max_retries": -1, "max_runtime": "00:00:00", "proc_schema": "public", "retry_period": "00:05:00", "initial_start": "00:05:00", "fixed_schedule": true, "schedule_interval": "01:00:00"}}
+   1002 | t         | {"job": {"owner": "super_user", "config": {"key": "value"}, "timezone": "America/Sao_Paulo", "proc_name": "custom_job_alter", "scheduled": true, "max_retries": -1, "max_runtime": "00:00:00", "proc_schema": "public", "retry_period": "00:05:00", "initial_start": "00:05:00", "fixed_schedule": false, "schedule_interval": "00:10:00"}}
+(2 rows)
```

Comment:
Would suggest to show the changed value only and some sort of "order of execution" to show that one is before the other. There is a lot of extra data above that is not relevant to what you're testing so would suggest to reduce that to avoid flakes and the need for irrelevant changes in later commits.

Reply:
Make sense... done!